### DISCUSSION TITLE
Hydrojet control in HA

### DISCUSSION THREAD

**Discussion by dasfliege** [visualapproach/WiFi-remote-for-Bestway-Lay-Z-SPA#479] [Q&A] [answered]
I have setup this project for my Maldives 2019 with 54173 pump. I have not connected it to the pump yet but am just trying to get everything represented in my HA dashboard. I am using Mosquitto integrated in HA as MQTT broker. Everything has been discovered automatically just fine except of the hydrojet control. Am i missing something, or is hydrojet just not integrated to MQTT yet?

**Comment by visualapproach** [comment, marked as answer]
Not integrated to MQTT but now it's on the TODO list 😸

**Reply by visualapproach** [reply]
It's now available in development branch ("beta" if updating from web UI)

**Reply by dasfliege** [reply]
Gona give it a try soon. Thx!

**Reply by dasfliege** [reply]
Hmm, it does not seem to work. At least Mosquitto is not automatically recognizing the Hydrojet entities. I also tried to delete and reconfigure everything.
But what seems a little bit curious to me, is that master and beta releases are the same when i update through the UI? Is this normal?

Current: 2023-03-27-2004
Master: 2023-03-27-2004
Beta: 2023-03-27-2004

**Reply by visualapproach** [reply]
It works here:
![image](https://user-images.githubusercontent.com/22987250/229151674-653c16d9-5c5e-46b3-9b98-d79db9c4b715.png)


Hardware config page: 54173
save
RESTART ESP
In HA go to config, tab devices. Look for layzspa. Click it and it should be there.


Regarding the releases - normal or not, in that case they were the same :-)
But not now.

**Reply by dasfliege** [reply]
MQTT in general is working fine, but i still have no hydrojet entity :-)
I don't see hydrojet in your HA screenshot as well?

**Comment by visualapproach** [comment]
![image](https://user-images.githubusercontent.com/22987250/229172535-598675ef-2a65-409e-b728-1ce24c4f1884.png)

**Reply by dasfliege** [reply]
Haha, should open my eyes sometime :-)
However, i do not have this item in my HA :-( Do you have a total of 40 entities for your device? I only have 38.
Gonna try some debugging myself  i guess

**Reply by visualapproach** [reply]
Get "MQTT explorer" and see what is beeing sent or not.

**Comment by dasfliege** [comment]
"HASJETS" equals true, but there is no sensor and no switch beeing sent. I actually don't have much knowledge about MQTT yet, so i might just missed some additional steps i have to perform??

![image](https://user-images.githubusercontent.com/31273000/229625858-5305e1f0-0894-485e-b702-82193f09dd44.png)

**Reply by dasfliege** [reply]
Any idea why the "jet" switch is not beeing sent? Thought it might be a problem because of the other problems i had with the UI, but now as i was able to set the correct pump model in de UI, the JET switch is still not beeing sent to HA.

**Reply by visualapproach** [reply]
Don't know. Try restarting esp from web ui and wait a minute.
Don't know if an update to latest fw will help.

**Reply by dasfliege** [reply]
Still nothing. Also not with with latest firmware. Tried to compeltely uninstall Mosquitto, reinstall, listen, restarted ESP, etc etc...
Well, at least everything else is working fine now, except of the hydros

**Reply by visualapproach** [reply]
@dasfliege Hi, I think the latest beta version will get you the jets-switch.

**Reply by dasfliege** [reply]
Ohh yes. It magically appeared now! Thanks a lot! :-)
